bullet_train-api/app/helpers/api/open_api_helper.rb

```diff
@@ -249,6 +249,9 @@ def customize_component!(original, custom, method_type = nil)
           if details["required"]
             original["required"] << property
             details.delete("required")
+          elsif details["required"] == false
+            original["required"].delete(property)
+            details.delete("required")
```

Comment:
```suggestion
          if details["required"]
            original["required"] << property
          else
            original["required"].delete(property)
          end
          # Always delete required because this attribute should only go
          # into the required array according to OpenAPI 3.1.
          details.delete("required")
```

As discussed, we can do this. ☝️ 

Additionally, we could start giving names to what's happening in this file, but maybe this is for a bigger refactoring later.

```ruby
def handle_required(...)
  required = details.delete("required")
  required ? original["required"] << property : original["required"].delete(property)
end
```